File: edexOsgi/build.edex/opt/esb/js/CatalogQuery.js
function CatalogQuery(plugin){
  this.cat = new MetadataCatalogQuery(plugin);
}

function _setDistinctField(name){
	this.cat.setDistinctField(name);
}

function _addReturnedField(name){
  this.cat.addReturnedField(name);
}

function _addConstraint(name, value,operand){
	if(arguments.length==2){
  		this.cat.addParameter(name, value);
  	}else{
  		this.cat.addParameter(name,value,operand);
  	}
}

function _execute(){
  return this.cat.execute();
}

CatalogQuery.prototype.addReturnedField = _addReturnedField;
CatalogQuery.prototype.addConstraint = _addConstraint;
CatalogQuery.prototype.execute = _execute;
CatalogQuery.prototype.setDistinctField = _setDistinctField;

File: edexOsgi/com.raytheon.edex.uengine/utility/edex_static/base/python/CatalogQuery.py
##
##

##
# This is a base file that is not intended to be overridden.
##

##
# uengine is deprecated and will be removed from the system soon. Migrate your
# apps to using the Data Access Framework (DAF).
##

#
# Query of data catalog
#
#    
#     SOFTWARE HISTORY
#    
#    Date            Ticket#       Engineer       Description
#    ------------    ----------    -----------    --------------------------
#    04/14/08                      njensen       Initial Creation.
#    
# 



from com.raytheon.edex.uengine.tasks.query import MetadataCatalogQuery
from com.raytheon.uf.common.message.response import ResponseMessageCatalog


class CatalogQuery():
    
    def __init__(self, pluginName):
        self.__cat = MetadataCatalogQuery(pluginName)
        
    def setDistinctField(self,field):
        self.__cat.setDistinctField(field)   
        
    def addReturnedField(self, name):
        self.__cat.addReturnedField(name,None)
        
    def addMaxReturnedField(self, name):
        self.__cat.addMaxReturnedField(name, None)
    
    def addConstraint(self, name, value, operand="="):   
        self.__cat.addParameter(name, value, operand)
    
    def execute(self):
        return self.__cat.execute()
    
    def executeWrapped(self):
        return ResponseMessageCatalog.wrap(self.__cat.execute())
